app/app.go

```diff
@@ -594,6 +600,12 @@ func (app *RegenApp) Name() string { return app.BaseApp.Name() }
 // BeginBlocker application updates every begin block
 func (app *RegenApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
 	resp := app.mm.BeginBlock(ctx, req)
+
+	// initialize global tendermint validator set before patch height
+	if app.votesInfo == nil {
+		app.votesInfo = req.LastCommitInfo.Votes
+	}
```

Comment:
This is wrong. The value of voteInfos will be non deterministic. Just set it always.
```suggestion
		app.votesInfo = req.LastCommitInfo.Votes
```